cmd/waku/server/rest/filter.go

```diff
@@ -0,0 +1,265 @@
+package rest
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/waku-org/go-waku/waku/v2/node"
+	"github.com/waku-org/go-waku/waku/v2/peermanager"
+	"github.com/waku-org/go-waku/waku/v2/protocol"
+	"github.com/waku-org/go-waku/waku/v2/protocol/filter"
+	"go.uber.org/zap"
+)
+
+type filterRequestId []byte
+
+func (r *filterRequestId) UnmarshalJSON(bodyBytes []byte) error {
+	body := strings.Trim(string(bodyBytes), `"`)
+	reqId, err := hex.DecodeString(body)
+	if err != nil {
+		return err
+	}
+	*r = reqId
+	return nil
+}
+
+func (r filterRequestId) String() string {
+	return hex.EncodeToString(r)
+}
+func (r filterRequestId) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf(`"%s"`, r.String())), nil
+}
+
+const filterv2Ping = "/filter/v2/subscriptions/{requestId}"
+const filterv2Subscribe = "/filter/v2/subscriptions"
+const filterv2SubscribeAll = "/filter/v2/subscriptions/all"
+
+// RelayService represents the REST service for WakuRelay
+type FilterService struct {
+	node *node.WakuNode
+
+	log *zap.Logger
+}
+
+// NewFilterService returns an instance of FilterService
+func NewFilterService(node *node.WakuNode, m *chi.Mux, log *zap.Logger) *FilterService {
+	s := &FilterService{
+		node: node,
+		log:  log.Named("filter"),
+	}
+
+	m.Get(filterv2Ping, s.ping)
+	m.Post(filterv2Subscribe, s.subscribe)
+	m.Delete(filterv2Subscribe, s.unsubscribe)
+	m.Delete(filterv2SubscribeAll, s.unsubscribeAll)
+
+	return s
+}
+
+// 400 for bad requestId
+// 404 when request failed or no suitable peers
+// 200 when ping successful
+func (s *FilterService) ping(w http.ResponseWriter, req *http.Request) {
+	var requestId filterRequestId
+	if err := requestId.UnmarshalJSON([]byte(chi.URLParam(req, "requestId"))); err != nil {
+		s.log.Error("bad request id", zap.Error(err))
+		writeResponse(w, &filterSubscriptionResponse{
+			RequestId:  requestId,
+			StatusDesc: "bad request id",
+		}, http.StatusBadRequest)
+		return
+	}
+
+	// selecting random peer that supports filter protocol
+	peerId := s.getRandomFilterPeer(req.Context(), requestId, w)
```

Comment:
I don't think this is a good idea, because if we are randomly selecting a peer from peerStore then there is a no gaurantee that we have a subscription with this peer which will inturn result in ping failure. 
Instead, we should take contentFilter as an argument for ping and use https://github.com/waku-org/go-waku/blob/2701a38b2aef3b1ece7b2aef7c8df13a88e069c5/waku/v2/protocol/filter/client.go#L332 to get the subscription and check if it is alive.

But since this looks like a change is required for REST endpoint, could be taken up later.